### PR TITLE
Add Api endpoints

### PR DESCRIPTION
**Pull request by yohannesdagnachew**
> ### Hello Reviewers 👋 👋
> ## In this project we did:
> ## Description
> In this PR, we added API endpoints for accessing our blog App.
> 
> ### General requirements
> * [x]  There are [no linter errors](https://github.com/microverseinc/linters-config).
> * [x]  Used [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
> * [x]  Professional documentation .
> 
> ### Ruby requirements
> * [x]  Followed list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).
> 
> ### Project requirements
> * [x]  Created an API endpoint to list all posts for a user.
> * [x]  Created an API endpoint to list all comments for a user's post.
> * [x]  Created an API endpoint to add a comment to a post.
> * [x]  Added authentication to your API endpoints.
> * [x]  The API endpoints receive JSON and respond JSON as well.
> 
> Thanks for reviewing our work!

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 Rails.application.routes.draw do
   devise_for :users
   resources :users, only: %i[index show] do
-    resources :posts, only: %i[index new create show]
+    resources :posts, only: %i[index new create show] do # rubocop:todo Lint/EmptyBlock
+    end
```
Comment (rustam-tolipov) [inline config/routes.rb:11]: - [ ] Well implementation. However I cannot see your api endpoints. Kindly add them as you described in controllers.


> Example:
> 
> http://localhost:3000/api/users/...
>  or ..api/users/1/posts..
>

Comment (rustam-tolipov) [reply config/routes.rb:11]: If you are using some external resources kindly provide screenshot or video explanation

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @yohannesdagnachew, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Great Design_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by peterrobert — APPROVED** [review]
Hi @yohannesdagnachew


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**